### PR TITLE
RGD: dummy atom in input structure is mishandled

### PR DESCRIPTION
**Pull request by jones-gareth**
#### What does this implement/fix? Explain your changes.

Currently RGD does not decompose correctly if an input structure contains a dummy atom.  For example, if a second RGD is performed on the groups from an earlier RGD.  Just to be clear- by submitting a fix I'm not endorsing the practice!

Before
 
Screenshot 2022-01-06 121328

After

Screenshot 2022-01-06 120844

Note: the difference in the core depiction is due to the improvements in master since the 21.09 release. The PR fixes the R2 and R4 groups.

### CODE REVIEW

**Comment by bp-kelley** [comment]
Well, that's not something I had planned on apparently!

File: Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp [hunk 1]
```diff
@@ -89,6 +89,17 @@ int RGroupDecomposition::add(const ROMol &inmol) {
   const bool addCoords = true;
   MolOps::addHs(mol, explicitOnly, addCoords);
 
+  // mark any wildcards in input molecule:
+  for (auto &atom : mol.atoms()) {
+    if (atom->getAtomicNum() == 0) {
+      atom->setIsotope(1000U);
```
Comment (bp-kelley) [inline Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp:95]: Might be better to use a tag here.  I for one have used 1000 isotopes as a tag in the past...

Comment (bp-kelley) [reply Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp:95]: something like
```
atom->setBoolProp("WASDUMMY", true);
```

**Review by bp-kelley — CHANGES_REQUESTED** [review]

File: Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp [hunk 2]
```diff
@@ -89,6 +89,18 @@ int RGroupDecomposition::add(const ROMol &inmol) {
   const bool addCoords = true;
   MolOps::addHs(mol, explicitOnly, addCoords);
 
+  // mark any wildcards in input molecule:
+  for (auto &atom : mol.atoms()) {
+    if (atom->getAtomicNum() == 0) {
+      atom->setProp("INPUT_DUMMY", true);
```
Comment (greglandrum) [inline Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp:95]: Definitely not required, but it would be better if you had a constexpr for `"INPUT_DUMMY"`

**Comment by greglandrum** [comment]
@jones-gareth : I just pushed a minor change to your fork which should fix the windows DLL builds.
Instead of doing that directly I should have created a PR against your fork, but I wasn't paying close enough attention to the web interface. Sorry for the direct commit.

**Review by greglandrum — APPROVED** [review]
I just have one minor suggestion. If you agree, please make the change, it LGTM either way.

File: Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp (hunk 2 above)
Comment (jones-gareth) [reply Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp:96]: Made that change

**Comment by greglandrum** [comment]
Assuming @bp-kelley agrees, I think we can merge this one.
Thanks @jones-gareth !

**Comment by greglandrum** [comment]
got an LGTM from @bp-kelley offline. Merging this.

**Pull request merged**